src/components/draw/DrawDirective.js

```diff
@@ -327,110 +559,227 @@ goog.require('ga_map_service');
           };
+          scope.deleteSelectedFeature = function(evt) {
+            if ((!evt || evt.which == 46) &&
+                select.getFeatures().getLength() > 0) {
+              if (layer.getSource().getFeatures().length == 1) {
+                scope.deleteAllFeatures();
+                return;
+              }
+              layer.getSource().removeFeature(select.getFeatures().item(0));
+              select.getFeatures().clear();
+            }
+          };
+          // Delete all features of the layer
+          scope.deleteAllFeatures = function() {
+            if (confirm($translate.instant('confirm_remove_all_features'))) {
+              select.getFeatures().clear();
+              layer.getSource().clear();
+              if (layer.adminId) {
+                scope.statusMsgId = '';
+                gaFileStorage.del(layer.adminId).then(function() {
+                  layer.adminId = undefined;
+                  layer.url = undefined;
+                  scope.adminShortenUrl = undefined;
+                  scope.userShortenUrl = undefined;
+                });
+              }
+              map.removeLayer(layer);
+            }
+          };
 
           scope.canExport = function() {
             return (layer) ? layer.getSource().getFeatures().length > 0 : false;
           };
 
-          scope.aToolIsActive = function() {
-            return !!lastActiveTool;
+
+
+          ////////////////////////////////////
+          // Shorten url stuff
+          ////////////////////////////////////
+          var updateShortenUrl = function(adminId) {
+
+            // For now we pass the long permalink otherwoise we need to
+            // regenerate the permalink on each map interaction
+            /*$http.get(scope.options.shortenUrl, {
+              params: {
+                url: gaPermalink.getHref().replace(
+                    gaUrlUtils.encodeUriQuery(layer.id, true), '') +
+                    '&adminId=' + adminId
+              }
+            }).success(function(data) {
+              scope.adminShortenUrl = data.shorturl;
+            });
+            $http.get(scope.options.shortenUrl, {
+              params: {
+                url: gaPermalink.getHref()
+              }
+            }).success(function(data) {
+              scope.userShortenUrl = data.shorturl;
+            });*/
+            scope.adminShortenUrl = gaPermalink.getHref().replace(
+                gaUrlUtils.encodeUriQuery(layer.id, true), '') +
+                '&adminId=' + adminId;
+            scope.userShortenUrl = gaPermalink.getHref();
           };
 
-          // Watchers
-          scope.$watch('isActive', function(active) {
-            if (active) {
-              activate();
+
+
+          ////////////////////////////////////
+          // Popup content management
+          ////////////////////////////////////
+          var popupTitlePrefix = 'draw_popup_title_';
+          var togglePopup = function(feature) {
```

Comment:
toggle in the name might confuse some people. Function does not really toggle popup, but activates if feature is present and deactivates when feature is not present. Minor though.